helix-term/src/commands.rs

```diff
@@ -3443,62 +3425,12 @@ fn view_mode(cx: &mut Context) {
     })
 }
 
-fn left_bracket_mode(cx: &mut Context) {
-    cx.on_next_key(move |cx, event| {
-        if let KeyEvent {
-            code: KeyCode::Char(ch),
-            ..
-        } = event
-        {
-            match ch {
-                'd' => goto_prev_diag(cx),
-                'D' => goto_first_diag(cx),
-                _ => (),
-            }
-        }
-    })
-}
-
-fn right_bracket_mode(cx: &mut Context) {
-    cx.on_next_key(move |cx, event| {
-        if let KeyEvent {
-            code: KeyCode::Char(ch),
-            ..
-        } = event
-        {
-            match ch {
-                'd' => goto_next_diag(cx),
-                'D' => goto_last_diag(cx),
-                _ => (),
-            }
-        }
-    })
+fn select_textobject_around(cx: &mut Context) {
+    select_textobject(cx, textobject::TextObject::Around);
 }
 
-use helix_core::surround;
-use helix_core::textobject;
-
-fn match_mode(cx: &mut Context) {
-    let count = cx.count;
```

Comment:
I tried adding this but doesn't seemed to make a difference for me.

Reply:
`on_next_key` always passes the count as None, so it's necessary for counts to work properly as of now:

https://github.com/helix-editor/helix/blob/a091e113bca9cef99f11373de04520ead906f3c0/helix-term/src/ui/editor.rs#L653-L664

Using the previous count when constructing `Context` if there's a registered `on_next_key` callback is the definitive fix anyway, which can be done later I suppose (count will be broken till then though).

Reply:
I did this but noticed nothing different.

Reply:
Oh it's handled in `on_next_key_mode` itself 👍🏾 :

https://github.com/helix-editor/helix/blob/009073fbfa3ef607363ec4e54ee347dcf1ac6398/helix-term/src/commands.rs#L80-L83